src/extension/openFileAtLocation.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {RemoteExtension} from "../common/remoteExtension";
+import {ReactNativeProjectHelper} from "../common/reactNativeProjectHelper";
+import {InternalErrorCode} from "../common/error/internalErrorCode";
+import {ErrorHelper} from "../common/error/errorHelper";
+import * as path from "path";
+
+/* Usage:
+...path\openFileAtLocation.js filename:lineNumber
+...path\openFileAtLocation.js filename
+*/
+
+try {
+    const fileInfo = process.argv[2].split(":");
+    const filename: string  = fileInfo[0];
+    let lineNumber: number = 1;
+    if (fileInfo.length >= 2) {
+        lineNumber = parseInt(fileInfo[1], 10);
+    }
+
+    getReactNativeWorkspaceForFile(filename).then(projectRootPath => {
+        const remoteExtensionServerPath = RemoteExtension.atProjectRootPath(projectRootPath);
+        return remoteExtensionServerPath.openFileAtLocation(filename, lineNumber);
+    }).done();
+} catch (e) {
+    throw ErrorHelper.getNestedError(e, InternalErrorCode.CommandFailed, "Unable to open file with args [" + process.argv.join(", ") + "]");
+}
+
+function getReactNativeWorkspaceForFile(file: string): Q.Promise<string> {
+    return getNameForRNParentWorkspace(path.dirname(file)).then(workspaceName => {
+        if (workspaceName) {
+            return workspaceName;
+        }
+        return Q.reject<string>("Could not find workspace for file " + file);
+    });
+}
+
+function getNameForRNParentWorkspace(dir: string): Q.Promise<string> {
+    const reactNativeProjectHelper = new ReactNativeProjectHelper(dir);
+    return reactNativeProjectHelper.isReactNativeProject().then(isRNProject => {
+        if (isRNProject) {
+            return dir;
+        }
+        if (dir === "" || dir === "." || dir === "/") {
```

Comment:
This should also account for `dir === path.dirname(dir)` to catch windows style directory roots